app/controllers/comments_controller.rb

```diff
@@ -1,4 +1,5 @@
 class CommentsController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
- **[OPTIONAL]**  Instead of repeating this line across all your controllers, it would be a good idea to move it to the application controller so that it is used across all controllers.

Reply:
yeah! thanks for the suggestion 😀 but I am unsure how to customize it for individual actions in the controller if I put it in the application controller 🤔

Reply:
Hi @batoolfatima2135 pardon the late reply. You can put it in the application controller. Then for individual actions where you don't want it, you use the `skip_before_action` helper. As an example...

```
skip_before_action :authenticate_user!, only: %i[show]
```
if you want to skip the authentication for only the `show` action.

or

```
skip_before_action :authenticate_user!, except: %i[show]
```
if you want to skip authentication for all actions, except `show` action. In this case only `show` action is authenticated

Reply:
Thanks, alot for the suggestion. I will definitely implement it. 😊